tests/components/mqtt/test_config_flow.py

```diff
@@ -597,3 +599,30 @@ async def test_options_bad_will_message_fails(hass, mock_try_connection):
         mqtt.CONF_BROKER: "test-broker",
         mqtt.CONF_PORT: 1234,
     }
+
+
+async def test_try_connection(hass, mqtt_mock):
```

Comment:
It would be better if `try_connection` was tested indirectly when used instead of by calling it directly.
Maybe modify the existing `mock_try_connection` so it wraps the real `try_connection`, then we can check both what is passed to `try_connection` + what it returned *and* that the config flow behaved as intended.

Reply:
done, still miss a test for the protocol parameter, is this option used from config flow?

Reply:
According to the code, the hassio plugin may set the protocol parameter:
https://github.com/home-assistant/core/blob/4051e2f518964dbdbbf326c984a656e0ca188fbf/homeassistant/components/mqtt/config_flow.py#L101-L108

Reply:
mocked the test to also include coverage for setting the protocol